src/runtime_src/core/pcie/tools/cloud-daemon/azure/azure.h

```diff
@@ -72,11 +72,27 @@ class AzureDev
         cfile.close();    
         return "";
     };
+    static std::vector<std::string> get_serial_number()
+    {
+        std::vector<std::string> ret;
```

Comment:
This ret is uninitialized and could possibly return some garbage value if no device found.